remote/bitbucketserver/internal/client.go

```diff
@@ -0,0 +1,234 @@
+	var token oauth.AccessToken
+	token.Token = AccessToken
+	client, err := Consumer.MakeHttpClient(&token)
+	log.Debug(fmt.Printf("Create client: %+v %s\n", token, url))
+	if err != nil {
+		log.Error(err)
+	}
+	return &Client{client, url, AccessToken}
+}
+
+func (c *Client) FindCurrentUser() (*model.User, error) {
+	CurrentUserIdResponse, err := c.client.Get(fmt.Sprintf(currentUserId, c.base))
+	if err != nil {
+		return nil, err
+	}
+	defer CurrentUserIdResponse.Body.Close()
+	bits, err := ioutil.ReadAll(CurrentUserIdResponse.Body)
+	if err != nil {
+		return nil, err
+	}
+	login := string(bits)
+
+	CurrentUserResponse, err := c.client.Get(fmt.Sprintf(pathUser, c.base, login))
+	if err != nil {
+		return nil, err
+	}
+	defer CurrentUserResponse.Body.Close()
+
+	contents, err := ioutil.ReadAll(CurrentUserResponse.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var user User
+	err = json.Unmarshal(contents, &user)
+	if err != nil {
+		return nil, err
+	}
+
+	ModelUser := &model.User{
+		Login:  login,
+		Email:  user.EmailAddress,
+		Token:  c.accessToken,
+		Avatar: avatarLink(user.EmailAddress),
+	}
+	log.Debug(fmt.Printf("User information: %+v\n", ModelUser))
+	return ModelUser, nil
+
+}
+
+func (c *Client) FindRepo(owner string, name string) (*model.Repo, error) {
+	urlString := fmt.Sprintf(pathRepo, c.base, owner, name)
+	response, err := c.client.Get(urlString)
+	if err != nil {
+		log.Error(err)
+	}
+	defer response.Body.Close()
+	contents, err := ioutil.ReadAll(response.Body)
+	bsRepo := BSRepo{}
+	err = json.Unmarshal(contents, &bsRepo)
+	if err != nil {
+		return nil, err
+	}
+	repo := &model.Repo{
+		Name:      bsRepo.Slug,
+		Owner:     bsRepo.Project.Key,
+		Branch:    "master",
+		Kind:      model.RepoGit,
+		IsPrivate: true, // Since we have to use Netrc it has to always be private :/
+		FullName:  fmt.Sprintf("%s/%s", bsRepo.Project.Key, bsRepo.Slug),
+	}
+
+	for _, item := range bsRepo.Links.Clone {
+		if item.Name == "http" {
+			uri, err := url.Parse(item.Href)
+			if err != nil {
+				return nil, err
+			}
+			uri.User = nil
+			repo.Clone = uri.String()
+		}
+	}
+	for _, item := range bsRepo.Links.Self {
+		if item.Href != "" {
+			repo.Link = item.Href
+		}
+	}
+	log.Debug(fmt.Printf("Repo: %+v\n", repo))
+	return repo, nil
+}
+
+func (c *Client) FindRepos() ([]*model.RepoLite, error) {
+	requestUrl := fmt.Sprintf(pathRepos, c.base, "1000")
+	log.Debug(fmt.Printf("request :%s", requestUrl))
+	response, err := c.client.Get(requestUrl)
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close()
+	contents, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	var repoResponse Repos
+	err = json.Unmarshal(contents, &repoResponse)
+	if err != nil {
+		return nil, err
+	}
+	log.Debug(fmt.Printf("repoResponse: %+v\n", repoResponse))
+	var repos = []*model.RepoLite{}
+	for _, repo := range repoResponse.Values {
+		repos = append(repos, &model.RepoLite{
+			Name:     repo.Slug,
```

Comment:
return a native bitbucket type and move the conversion to a `convert.go` file in the parent package